interfaces/builtin/ofono.go

```diff
@@ -0,0 +1,308 @@
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var ofonoPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the ofono service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# ofono puts ppp on top of the tun device
+/dev/net/tun rw,
+# to create ppp network interfaces
+capability net_admin,
+
+# To check present devices
+/run/udev/data/* r,
+/sys/bus/usb/devices/ r,
+# FIXME snapd should be querying udev and adding the /sys and /run/udev accesses
+# that are assigned to the snap, but we are not there yet.
+/sys/bus/usb/devices/** r,
+
+# To get current seat preferences
+/run/systemd/seats{,/*} r,
+
+# Access to modem ports
+# FIXME snapd should be more dynamic to avoid conflicts between snaps trying to
+# access same ports.
+/dev/tty[^0-9]* rw,
+/dev/cdc-* rw,
+
+network netlink,
+network netlink raw,
+network netlink dgram,
+network bridge,
+network inet,
+network inet6,
+network packet,
+network bluetooth,
+
+include <abstractions/nameservice>
+
+# DBus accesses
+include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={Request,Release}Name
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.ofono",
+`)
+
+var ofonoConnectedSlotAppArmor = []byte(`
+# Allow service to interact with connected clients
+
+# Allow traffic to/from our interfaces. The path depends on the modem plugin,
+# and is arbitrary.
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var ofonoConnectedPlugAppArmor = []byte(`
+# Description: Allow using Ofono service. Reserved because this gives
+#  privileged access to the Ofono service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to ofono services
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`)
+
+var ofonoPermanentSlotSecComp = []byte(`
+# Description: Allow operating as the ofono service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+# TODO: add ioctl argument filters when seccomp arg filtering is implemented
```

Comment:
Comment was a left over, removed. Added a comment to clarify what are the calls for.